<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModelMetricsWidget</class>
 <widget class="QWidget" name="ModelMetricsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Model Metrics</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>6</number>
   </property>
   <property name="leftMargin">
    <number>9</number>
   </property>
   <property name="topMargin">
    <number>9</number>
   </property>
   <property name="rightMargin">
    <number>9</number>
   </property>
   <property name="bottomMargin">
    <number>9</number>
   </property>
   <item>
    <widget class="QGroupBox" name="metricsGroupBox">
     <property name="title">
      <string>Model Performance Metrics</string>
     </property>
     <layout class="QVBoxLayout" name="metricsLayout">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QLabel" name="modelTypeLabel">
        <property name="text">
         <string>Model Type: Not Set</string>
        </property>
        <property name="styleSheet">
         <string>QLabel { font-weight: bold; color: #333; }</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="separatorLine">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="metricsStackedWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="binaryClassificationPage">
         <layout class="QFormLayout" name="binaryClassificationLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="horizontalSpacing">
           <number>10</number>
          </property>
          <property name="verticalSpacing">
           <number>8</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="accuracyTextLabel">
            <property name="text">
             <string>Accuracy:</string>
            </property>
            <property name="toolTip">
             <string>Overall correctness: (TP + TN) / (TP + TN + FP + FN)</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="accuracyValueLabel">
            <property name="text">
             <string>--</string>
            </property>
            <property name="styleSheet">
             <string>QLabel { font-weight: bold; color: #2E8B57; }</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="sensitivityTextLabel">
            <property name="text">
             <string>Sensitivity (Recall):</string>
            </property>
            <property name="toolTip">
             <string>True Positive Rate: TP / (TP + FN)</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="sensitivityValueLabel">
            <property name="text">
             <string>--</string>
            </property>
            <property name="styleSheet">
             <string>QLabel { font-weight: bold; color: #2E8B57; }</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="specificityTextLabel">
            <property name="text">
             <string>Specificity:</string>
            </property>
            <property name="toolTip">
             <string>True Negative Rate: TN / (TN + FP)</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="specificityValueLabel">
            <property name="text">
             <string>--</string>
            </property>
            <property name="styleSheet">
             <string>QLabel { font-weight: bold; color: #2E8B57; }</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="diceScoreTextLabel">
            <property name="text">
             <string>Dice Score (F1):</string>
            </property>
            <property name="toolTip">
             <string>Harmonic mean of precision and recall: 2*TP / (2*TP + FP + FN)</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="diceScoreValueLabel">
            <property name="text">
             <string>--</string>
            </property>
            <property name="styleSheet">
             <string>QLabel { font-weight: bold; color: #2E8B57; }</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="Line" name="confusionMatrixSeparator">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QLabel" name="confusionMatrixLabel">
            <property name="text">
             <string>Confusion Matrix:</string>
            </property>
            <property name="styleSheet">
             <string>QLabel { font-weight: bold; }</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0" colspan="2">
           <widget class="QLabel" name="confusionMatrixDisplayLabel">
            <property name="text">
             <string>No data available</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="styleSheet">
             <string>QLabel { 
    font-family: 'Courier New', monospace; 
    background-color: #f5f5f5; 
    border: 1px solid #ddd; 
    padding: 8px; 
    border-radius: 4px;
}</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="noMetricsPage">
         <layout class="QVBoxLayout" name="noMetricsLayout">
          <item>
           <spacer name="topSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="noMetricsLabel">
            <property name="text">
             <string>No model metrics available.
Train a model to see performance metrics.</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="styleSheet">
             <string>QLabel { 
    color: #666; 
    font-style: italic;
    font-size: 11pt;
}</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="bottomSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 